### ISSUE TITLE
Contextual "And" handling

### ISSUE THREAD

**Issue by bbqsrc** [Ahmed-Ali/Cucumberish#41]
Cucumberish currently parses the gherkin literally, so it makes it hard for me to compose tests where a test might be written `Given I am in a state` or `And I am in a state`. I will have to duplicate the entry with a `Given("I am in a state")` and an `And("I am in a state")` as the `And` will never find the `Given`.

Other cucumber implementations understand the context of the `And` based on the last used token. Is this something that could be easily added?

**Comment by TitouanVanBelle** [comment]
The only solution the moment is to write something like this

```
    Match(@[@"When", @"And"], @"I am in a state", ^(NSArray<NSString *> *args, NSDictionary *userInfo) {
    });
```

But I agree that this makes the step definition less readable. I would suggest the following implementation

```
void Given(NSString * definitionString, CCIStepBody body)
{
    addDefinition(definitionString, body, @"Given");
    addDefinition(definitionString, body, @"And");
}
void When(NSString * definitionString, CCIStepBody body)
{
    addDefinition(definitionString, body, @"When");
    addDefinition(definitionString, body, @"And");
}
void Then(NSString * definitionString, CCIStepBody body)
{
    addDefinition(definitionString, body, @"Then");
    addDefinition(definitionString, body, @"And");
}
void But(NSString * definitionString, CCIStepBody body)
{
    addDefinition(definitionString, body, @"But");
}
```

And I would also deprecate the And() method.

**Comment by brentleyjones** [comment]
I would love to have the contextual support as well. I've personally implemented helper methods that essentially does what @TitouanVanBelle does above with `Match()`.

Anyone willing to take a shot at implementing it in a PR?

**Comment by TitouanVanBelle** [comment]
Took a shot a it this morning. Let me know what you think.

**Comment by TitouanVanBelle** [comment]
@brentleyjones Any idea when this will be released?

**Comment by brentleyjones** [comment]
I'll work with @Ahmed-Ali on getting a release ready.

**Comment by brentleyjones** [comment]
This has been released in v1.2.0. Cocoa pods will be updated as soon as @Ahmed-Ali pushes it to trunk.
